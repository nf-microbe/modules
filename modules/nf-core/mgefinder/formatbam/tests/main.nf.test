nextflow_process {

    name "Test Process MGEFINDER_FORMATBAM"
    script "../main.nf"
    process "MGEFINDER_FORMATBAM"

    tag "modules"
    tag "modules_nfcore"
    tag "mgefinder"
    tag "mgefinder/formatbam"

    test("bam") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("/workspaces/modules/.nf-test/tests/dd0caa451f454f495f6763b30412274c/work/1f/292e248cc8ec577c729c6a46ac2dbf/test.bam", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bam - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    []
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
