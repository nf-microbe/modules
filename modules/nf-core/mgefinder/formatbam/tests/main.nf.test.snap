{
    "bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.mgefinder.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.mgefinder.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T21:58:00.88243674"
    },
    "bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,c0e6a077a10a3153be027e7ee8f8c1e9",
                            "test.mgefinder.bam.bai:md5,ead90b7bbca94b50608fc61ab348cf78"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.mgefinder.bam:md5,c0e6a077a10a3153be027e7ee8f8c1e9",
                            "test.mgefinder.bam.bai:md5,ead90b7bbca94b50608fc61ab348cf78"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,68fb8eb0a87c3128ae275e2103e176f1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T21:56:53.727096522"
    }
}